### PR TITLE
Add py39 target in Black's configuration, bump Black to 21.5b2

### PR DESCRIPTION
**Pull request by fangchenli**

### CODE REVIEW

**Comment by fangchenli** [comment]
The py39 option is only on Black's master. Close for now.

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -9,7 +9,7 @@ repos:
     -   id: absolufy-imports
         files: ^pandas/
 -   repo: https://github.com/python/black
-    rev: 20.8b1
+    rev: 21.5b0
```
Comment (fangchenli) [inline .pre-commit-config.yaml:12]: The latest is `21.5b1`. But it's not on conda yet.

**Comment by MarcoGorelli** [comment]
Thanks @fangchenli - check #41160, there's some comments there about some things which probably should be manually fixed up

**Comment by jreback** [comment]
can you rebase

**Comment by jreback** [comment]
cc @MarcoGorelli are we generally ok with this?

**Comment by MarcoGorelli** [comment]
Looks good to me, seems that you made these changes manually:
```diff
diff --git a/pandas/_libs/algos.pyi b/pandas/_libs/algos.pyi
index a2d028250e..96dfd81153 100644
--- a/pandas/_libs/algos.pyi
+++ b/pandas/_libs/algos.pyi
@@ -50,15 +50,15 @@ def nancorr(
     mat: np.ndarray,  # const float64_t[:, :]
     cov: bool = False,
     minp=None,
-) -> np.ndarray: ...
+) -> np.ndarray: ...  # ndarray[float64_t, ndim=2]
 def nancorr_spearman(
     mat: np.ndarray,  # ndarray[float64_t, ndim=2]
     minp: int = 1,
-) -> np.ndarray: ...
+) -> np.ndarray: ...  # ndarray[float64_t, ndim=2]
 def nancorr_kendall(
     mat: np.ndarray,  # ndarray[float64_t, ndim=2]
     minp: int = 1,
-) -> np.ndarray: ...
+) -> np.ndarray: ...  # ndarray[float64_t, ndim=2]
 
 # ----------------------------------------------------------------------
 
diff --git a/pandas/_libs/tslibs/conversion.pyi b/pandas/_libs/tslibs/conversion.pyi
index 10c8a819b9..e74a56a519 100644
--- a/pandas/_libs/tslibs/conversion.pyi
+++ b/pandas/_libs/tslibs/conversion.pyi
@@ -12,7 +12,7 @@ class OutOfBoundsTimedelta(ValueError): ...
 
 def precision_from_unit(
     unit: str,
-) -> tuple[int, int,]: ...  # int64_t
+) -> tuple[int, int,]: ...  # (int64_t, _)
 def ensure_datetime64ns(
     arr: np.ndarray,  # np.ndarray[datetime64[ANY]]
     copy: bool = ...,
@@ -23,5 +23,5 @@ def ensure_timedelta64ns(
 ) -> np.ndarray: ...  # np.ndarray[timedelta64ns]
 def datetime_to_datetime64(
     values: np.ndarray,  # np.ndarray[object]
-) -> tuple[np.ndarray, tzinfo | None,]: ...  # np.ndarray[dt64ns]
+) -> tuple[np.ndarray, tzinfo | None,]: ...  # (np.ndarray[dt64ns], _)
 def localize_pydatetime(dt: datetime, tz: tzinfo | None) -> datetime: ...
diff --git a/pandas/_libs/window/indexers.pyi b/pandas/_libs/window/indexers.pyi
index 2dea936222..f65deae6fd 100644
--- a/pandas/_libs/window/indexers.pyi
+++ b/pandas/_libs/window/indexers.pyi
@@ -7,4 +7,7 @@ def calculate_variable_window_bounds(
     center: bool,
     closed: str | None,
     index: np.ndarray,  # const int64_t[:]
-) -> tuple[np.ndarray, np.ndarray,]: ...  # np.ndarray[np.int64]  # np.ndarray[np.int64]
+) -> tuple[
+    np.ndarray,
+    np.ndarray,
+]: ...  # (np.ndarray[np.int64]  # np.ndarray[np.int64])
```

which I believe address @jbrockmendel 's comments on the other PR

File: pandas/_libs/algos.pyi [hunk 2]
```diff
@@ -92,58 +75,40 @@ def nancorr_kendall(
...
-) -> None:
-    ...
-
-
+) -> None: ...
 def backfill(
     old: np.ndarray,  # ndarray[algos_t]
     new: np.ndarray,  # ndarray[algos_t]
     limit=None,
-) -> np.ndarray:  # np.ndarray[np.intp, ndim=1]
```
Comment (jbrockmendel) [inline pandas/_libs/algos.pyi:125]: why did this get lost?

File: pandas/_libs/algos.pyi [hunk 3]
```diff
@@ -92,58 +75,40 @@ def nancorr_kendall(
...
 def backfill_2d_inplace(
     values: np.ndarray,  # algos_t[:, :]
-    mask: np.ndarray,    # const uint8_t[:, :]
+    mask: np.ndarray,  # const uint8_t[:, :]
     limit=None,
 ) -> None: ...
-
-
 def is_monotonic(
-    arr: np.ndarray,  # ndarray[algos_t, ndim=1]
```
Comment (jbrockmendel) [inline pandas/_libs/algos.pyi:143]: the idea with the current formatting is to have a single arg on the line so it is super-obvious which argument the comment refers to

File: pandas/_libs/hashtable.pyi [hunk 4]
```diff
@@ -124,37 +116,31 @@ class HashTable:
...
-
     # TODO: `item` type is subclass-specific
     def get_item(self, item): ...  # TODO: return type?
     def set_item(self, item) -> None: ...
-
     # FIXME: we don't actually have this for StringHashTable or ObjectHashTable?
     def map(
         self,
-        keys: np.ndarray,     # np.ndarray[subclass-specific]
-        values: np.ndarray,   # const int64_t[:] values
```
Comment (jbrockmendel) [inline pandas/_libs/hashtable.pyi:136]: typo the "values" shouldnt be in the comment

File: pandas/_libs/hashtable.pyi [hunk 5]
```diff
@@ -211,46 +195,33 @@ class UInt64HashTable(HashTable): ...
...
     values: np.ndarray,  # const int64_t[:] values
     dropna: bool,
 ) -> np.ndarray: ...  # np.ndarray[np.int64]
-
 def value_count_int64(
     values: np.ndarray,  # const int64_t[:]
     dropna: bool,
-) -> tuple[
-    np.ndarray,  # np.ndarray[np.int64]
-    np.ndarray,  # np.ndarray[np.int64]
```
Comment (jbrockmendel) [inline pandas/_libs/hashtable.pyi:235]: same as above re this being multiple lines

File: pandas/_libs/hashtable.pyi [hunk 6]
```diff
@@ -211,46 +195,33 @@ class UInt64HashTable(HashTable): ...
...
-
 def value_count(
     values: np.ndarray,
     dropna: bool,
-) -> tuple[
-    np.ndarray,
-    np.ndarray,  # np.ndarray[np.int64]
-]: ...
-
+) -> tuple[np.ndarray, np.ndarray,]: ...  # np.ndarray[np.int64]
```
Comment (jbrockmendel) [inline pandas/_libs/hashtable.pyi:224]: this is ambiguous which ndarray the comment refers to

File: pandas/_libs/reshape.pyi [hunk 7]
```diff
@@ -1,19 +1,14 @@
...
 ) -> None: ...
-
-
 def explode(
     values: np.ndarray,  # np.ndarray[object]
-) -> tuple[
-    np.ndarray,  # np.ndarray[object]
-    np.ndarray,  # np.ndarray[np.int64]
-]: ...
+) -> tuple[np.ndarray, np.ndarray,]: ...  # np.ndarray[object]  # np.ndarray[np.int64]
```
Comment (jbrockmendel) [inline pandas/_libs/reshape.pyi:14]: this case isnt ambiguous, but it is less obvious

File: pandas/_libs/tslibs/conversion.pyi [hunk 8]
```diff
@@ -12,30 +12,16 @@ class OutOfBoundsTimedelta(ValueError): ...
 
 def precision_from_unit(
     unit: str,
-) -> tuple[
-    int,  # int64_t
-    int,
-]: ...
-
-
+) -> tuple[int, int,]: ...  # (int64_t, _)
```
Comment (jbrockmendel) [inline pandas/_libs/tslibs/conversion.pyi:15]: havent seen this pattern elsewhere?

File: pandas/_libs/tslibs/timezones.pyi [hunk 9]
```diff
@@ -13,20 +13,17 @@ import numpy as np
...
 
-
 def tz_standardize(tz: tzinfo) -> tzinfo: ...
-
 def tz_compare(start: Optional[tzinfo], end: Optional[tzinfo]) -> bool: ...
-
 def infer_tzinfo(
-    start: Optional[datetime], end: Optional[datetime],
+    start: Optional[datetime],
+    end: Optional[datetime],
```
Comment (jbrockmendel) [inline pandas/_libs/tslibs/timezones.pyi:20]: Optional -> datetime | None

File: pandas/_libs/algos.pyi (hunk 2 above)
Comment (MarcoGorelli) [reply pandas/_libs/algos.pyi:125]: bug in `black` I guess - maybe let's open an issue with them about this before merging

**Review by MarcoGorelli — CHANGES_REQUESTED** [review]

**Review by simonjayhawkins — CHANGES_REQUESTED** [review]
Thanks @fangchenli can we do this before 1.3.0rc0 or after 1.3.0. updating master during the rc period will potentially make auto backports fail.

File: pandas/_libs/tslibs/conversion.pyi (hunk 8 above)
Comment (fangchenli) [reply pandas/_libs/tslibs/conversion.pyi:15]: can we use this pattern `# (comment for A,  comment for B)` for the return type of `tuple[A, B]`? It's more explicit than the one reformated by `black`

**Comment by fangchenli** [comment]
> Thanks @fangchenli can we do this before 1.3.0rc0 or after 1.3.0. updating master during the rc period will potentially make auto backports fail.

I'll try to finish this before 1.3.0rc0.

**Comment by jreback** [comment]
consensus is to do this before the RC, @fangchenli can you merge master

**Comment by jreback** [comment]
thanks @fangchenli (we might need to add this to the development changes in whatsnew but can followup on that)

**Pull request merged**